README.md

```diff
@@ -118,26 +118,250 @@ services:
+
+You can view the status of one or more services by using `pebble services`:
+
+```
+$ pebble services srv1       # show status of a single service
+Service  Startup  Current
+srv1     enabled  active
+
+$ pebble services            # show status of all services
+Service  Startup   Current
+srv1     enabled   active
+srv2     disabled  inactive
+```
+
+The "Startup" column shows whether this service should be automatically started when Pebble starts ("enabled" means auto-start, "disabled" means don't auto-start).
+
+The "Current" column shows the current status of the service, and can be one of the following:
+
+* `active`: starting or running
+* `inactive`: not yet started, being stopped, or stopped
+* `backoff`: in a [backoff-restart loop](#service-auto-restart)
+* `error`: in an error state (currently: exited and exit action is "ignore")
+
+Implementation note: Pebble's service manager uses a [state machine](https://github.com/canonical/pebble/blob/a1f52239079e58cb6e3b6c7472ef5438b91eefab/internal/overlord/servstate/handlers.go#L67-L82) to manage the life cycle of a service. You can see these internal states and their transitions in the [state diagram](https://raw.githubusercontent.com/canonical/pebble/master/internal/overlord/servstate/state-diagram.svg).
+
+### Service dependencies
+
+Pebble takes service dependencies into account when starting and stopping services. Before the service manager starts a service, Pebble first starts the services that service depends on (configured with `required`). Conversely, before stopping a service, Pebble first stops services that depend on that service.
+
+For example, if service `nginx` requires `logger`, `pebble start nginx` will start `logger` and then start `nginx`. Running `pebble stop logger` will stop `nginx` and then `logger`; however, running `pebble stop nginx` will only stop `nginx` (`nginx` depends on `logger`, not the other way around).
+
+If multiple dependencies need to be started at once, they're started in order according to the `before` and `after` configuration: `before` is a list of services that must be started before this one (but it doesn't `require` them). Or if it's easier to specify the other way around, `after` is a list of services that must be started after this one.
+
+If the configuration of `requires`, `before`, and `after` for a service results in a cycle or "loop", an error will be returned when attempting to start or stop the service.
+
+### Service auto-restart
+
+Pebble's service manager automatically restarts services that exit unexpectedly. By default, this is done whether the exit code is zero or non-zero, but you can change this using the `on-success` and `on-failure` fields in a configuration layer. The possible values for these fields are:
```

Comment:
Do we explain configuration layers before talking about them?

Reply:
We do: configuration layers are discussed in the second main section of the doc (that I presume you wrote originally), "Layer configuration examples".

Reply:
Sounds good, thank you.